test/test_spec.py

```diff
@@ -1036,11 +1176,53 @@ class GreaterThanSpec:
 
             return GreaterThanSpec
 
+        def _get_no_extra_gt_10_spec():
+            class GreaterThanSpec:
+                key = Checker([gt_check], GT_CHECK=10)
+
+            return GreaterThanSpec
+
         ok_data = dict(key=11)
         assert validate_data_spec(ok_data, _get_gt_10_spec())
+        assert validate_data_spec(ok_data, _get_no_extra_gt_10_spec())
 
         nok_data = dict(key=10)
         assert is_something_error(ValueError, validate_data_spec, nok_data, _get_gt_10_spec())
+        assert is_something_error(ValueError, validate_data_spec, nok_data, _get_no_extra_gt_10_spec())
+
+
+class TestCheckKeyword(unittest.TestCase):
+    def test_check_keyword_must_upper_case(self):
+        assert Checker([STR], WHAT_EVER=True, MUST_BE_UPPER={'1': 1, '2': 2}, CASE=[1, 2])
+
+        with self.assertRaises(TypeError):
+            Checker([STR], WHAT_eVER=True)
+
+    def test_blacklist_check_keyword(self):
+        with self.assertRaises(TypeError):
+            Checker([STR], ALLOW_NONE=True)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], OP='SOME_OP')
+
+        with self.assertRaises(TypeError):
+            Checker([STR], OPTIONAL=True)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], EXTRA=dict())
+
+    def test_repeated_check_keyword(self):
+        with self.assertRaises(TypeError):
+            Checker([STR], optional=True, OPTIONAL=False)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], op=CheckerOP.ANY, OP=CheckerOP.ALL)
+
+        with self.assertRaises(TypeError):
+            Checker([STR], allow_none=True, ALLOW_NONE=False)
```

Comment:
nit: I think it is hard to identify the diff between blacklist and repeated cases

Reply:
yeah, you're right, I think I can either make the warning more informative or throw different kinds of errors.
Or both, I'll address it. thanks